tests/test_architectures.py

```diff
@@ -87,6 +92,38 @@ def __init__(self):
     mock_proxy_model = mock_model.create_model()
     return mock_proxy_model
 
+def _run_decomposition_test(primal_model_class, num_components):
+    X = _load_decomposition_data()
+    primal_model = primal_model_class(n_components= num_components)
+    proxy_model = _mock_dope(primal_model)
+    sess= tf.Session()
+
+    tf_trans_x = proxy_model.fit_transform(X)
+    tf_sigma = proxy_model.singular_values_ 
+    tf_U = tf_trans_x/tf_sigma
+    tf_V= proxy_model.components_
+    tf_approx_recon =  sess.run(tf.matmul(tf_U, tf.matmul(tf.linalg.diag(tf_sigma), tf_V)))#, adjoint_b=True) v.T in arch-line#161
+    #tf_approx_recon= np.dot(tf_U, np.dot(np.diag(tf_sigma), tf_V))#Since tf_U/tf_V are not tensors anymore
+
+    skl_trans_x = primal_model.fit_transform(X)
+    skl_sigma = primal_model.singular_values_    
+    skl_U = skl_trans_x/skl_sigma
+    skl_V= primal_model.components_
+    skl_approx_recon = np.dot(skl_U, np.dot(np.diag(skl_sigma), skl_V))
+
+    result = np.allclose(tf_approx_recon, skl_approx_recon)
+    _, p_value = stats.ttest_rel(tf_sigma, skl_sigma)
+
+    return result, p_value
+
+def test_svd_reconstruction():
+    result, _ = _run_decomposition_test(TruncatedSVD, 10)
+    assert result is True
+
+def test_svd_sigma_vals():
```

Comment:
what is the null hyptothesis of a t-test. typically, null is, there is no difference between means. if p < 0.05, then you reject the null (that there is no difference). 

if the above convention is true, then this t-test incorrectly accepts a test (that the mean of the eigen values between the two implementations is the same when in fact they are different). If effectively flips the decision in the opposite direction. 

in anycase, t-test does not make much sense here.

Reply:
verify (and if fix the t-test p-value part)